### PR TITLE
Alias add implemented, Sample Usage: set w=wifi

### PR DESCRIPTION
**Pull request by dbCooper95**
Alias add feature Implemented
Sample Usage: set w=wifi, set p=launch perplexy

Alias remove feature:
Sample usage: unset w

### CODE REVIEW

**Comment by crearo** [comment]
I get the `ArgType` `alias_remove`. However, do we need `ArgType` `alias_add`?
Remember, the only reason we want to know what type of arg it is is to give the user suggestions.

**Comment by dbCooper95** [comment]
> I get the `ArgType` `alias_remove`. However, do we need `ArgType` `alias_add`?
> Remember, the only reason we want to know what type of arg it is is to give the user suggestions.

The command alias_add has two arguments. I will create a default argumentType for first one, which all arguments with no suggestions can use. We would need suggestions for the second argument. The suggestions would be the list the existing commands. So will create a new argumentType for that too.

**Comment by crearo** [comment]
Yes, that makes sense. Can you please do this in this PR itself?

File: app/src/main/java/com/contextgenesis/chatlauncher/manager/input/InputParser.java [hunk 1]
```diff
@@ -76,32 +76,21 @@ public InputMessage parse(String input) {
...
                         String aliasName = args[i];
                         // if the alias doesnt exist
                         if (!aliasManager.containsAlias(aliasName.trim())) {
                             return InputMessage.invalidMessage(trimInput, commandType);
                         }
                         break;
+                    case COMMAND:
+                        // command needs to be validated
+                        String command = args[i].trim();
+                        InputMessage commandInputMessage = parse(command);
```
Comment (crearo) [inline app/src/main/java/com/contextgenesis/chatlauncher/manager/input/InputParser.java:89]: nice

File: app/src/main/java/com/contextgenesis/chatlauncher/manager/input/InputParser.java [hunk 2]
```diff
@@ -113,6 +102,15 @@ public InputMessage parse(String input) {
                             return InputMessage.invalidMessage(trimInput, commandType);
                         }
                         break;
+                    case NO_SUGGEST:
+                        // for commandType alias_add, aliasName should always be unique
```
Comment (crearo) [inline app/src/main/java/com/contextgenesis/chatlauncher/manager/input/InputParser.java:106]: clever catch

File: app/src/main/java/com/contextgenesis/chatlauncher/manager/alias/AliasManager.java [hunk 3]
```diff
@@ -13,25 +15,30 @@
 
     private final Set<Alias> aliasSet;
```
Comment (crearo) [inline app/src/main/java/com/contextgenesis/chatlauncher/manager/alias/AliasManager.java:19]: My only problem is, do we need this?

**Review by crearo — COMMENTED** [review]
Minor comments.

File: app/src/main/java/com/contextgenesis/chatlauncher/manager/alias/AliasManager.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public String removeAlias(String aliasName) {
+        Alias rmAlias = getAlias(aliasName);
+        aliasRepository.delete(aliasName);
+        aliasSet.remove(rmAlias);
+        return "Alias remove successful";
+    }
+
+    public boolean containsAlias(String alias) {
+        return aliasSet.contains(new Alias(alias, ""));
```
Comment (crearo) [inline app/src/main/java/com/contextgenesis/chatlauncher/manager/alias/AliasManager.java:46]: basically, this can be a database call directly

**Review by crearo — APPROVED** [review]
lgtm. there still are changes that we can fix in a separate PR. 
I've sprinkled some todos in the codebase for this.

**Pull request merged**